Source/WebGPU/WebGPU/QuerySet.h

```diff
@@ -59,24 +59,26 @@ class QuerySet : public WGPUQuerySetImpl, public RefCounted<QuerySet> {
     void destroy();
     void setLabel(String&&);
 
-    bool isValid() const { return m_queryCount > 0; }
+    bool isValid() const { return static_cast<bool>(m_visibilityBuffer) || static_cast<bool>(m_visibilityBuffer); }
 
     Device& device() const { return m_device; }
     Vector<MTLTimestamp> resolveTimestamps() const;
-    uint32_t queryCount() const { return m_queryCount; }
-    WGPUQueryType queryType() const;
+    uint32_t count() const { return m_count; }
+    WGPUQueryType type() const { return m_type; }
     id<MTLBuffer> visibilityBuffer() const { return m_visibilityBuffer; }
     id<MTLCounterSampleBuffer> counterSampleBuffer() const { return m_timestampBuffer; }
 
 private:
-    QuerySet(id<MTLBuffer>, Device&);
-    QuerySet(id<MTLCounterSampleBuffer>, Device&);
+    QuerySet(id<MTLBuffer>, uint32_t, WGPUQueryType, Device&);
+    QuerySet(id<MTLCounterSampleBuffer>, uint32_t, WGPUQueryType, Device&);
     QuerySet(Device&);
 
     const Ref<Device> m_device;
+    // FIXME: Can we use a variant for these two resources?
     id<MTLBuffer> m_visibilityBuffer { nil };
     id<MTLCounterSampleBuffer> m_timestampBuffer { nil };
-    uint32_t m_queryCount { 0 };
+    uint32_t m_count { 0 };
+    WGPUQueryType m_type { WGPUQueryType_Occlusion };
```

Comment:
Do we really need a state variable for this? I mean it is fine but it seems unnecessary especially since it seems pipeline statistics was removed from the web api

Reply:
I think we should implement pipeline statistics some time in the future (at a low priority). It's not difficult to do; I think we might as well just do it eventually.

Reply:
(I also think, even if pipeline statistics gets removed, there will be more query types added in the future, so handling n types is more forward-looking than assuming there are 2

Reply:
That’s fine with me though personally I think even having counter sampler queries in the API is not the best idea given the lack of consistent support across platforms / architectures